### ISSUE TITLE
Add prune section to Gopkg.toml 

### ISSUE THREAD

**Issue by praveenkumar** [minishift/minishift#2834]
As of now, all the packages which are part of our vendor have there tests also in and README, docs ..etc. 

Gopkg.toml can be updated to use `prune` section to make the only used library to part of vendor. 

```
# With prune
$ du -sh vendor/
27M	vendor/

# Without prune
$ du -sh vendor/
112M	vendor/
```

**Comment by LalatenduMohanty** [comment]
https://golang.github.io/dep/docs/Gopkg.lock.html#pruneopts

**Comment by LalatenduMohanty** [comment]
Resolved #2835
